libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -48,6 +67,19 @@ class NavigationView : ConstraintLayout, LifecycleObserver {
             )
         }
     }
+    private val distanceFormatter = MapboxDistanceFormatter(
```

Comment:
I don't think the NavigationView should be creating all of these dependencies for other classes like view models. Just because we're not using a dependency injection framework doesn't mean the NavigationView should take on this role. The providing of dependencies should be abstracted from the view models as well as the NavigationView. No need to hold up this PR for this issue. I'll propose an alternative to this separately.

Reply:
In fact the view model factories would be a good place to move these to.

Reply:
Yep I agree and I had the same intention to improve this later and focus on the item relevant to this PR. Thanks!